service/worker/scanner/history/scavenger.go

```diff
@@ -250,12 +251,27 @@ func (s *Scavenger) filterTask(
 	}
 	shardID := common.WorkflowIDToHistoryShard(namespaceID, workflowID, s.numShards)
 
+	branchToken, err := serialization.HistoryBranchToBlob(&persistencepb.HistoryBranch{
+		TreeId:    branch.BranchInfo.TreeId,
+		BranchId:  branch.BranchInfo.BranchId,
+		Ancestors: branch.BranchInfo.Ancestors,
+	})
```

Comment:
I don't think we need to copy each field?

```suggestion
	branchToken, err := serialization.HistoryBranchToBlob(branch.BranchInfo)
```